x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -50,6 +51,12 @@ public static DataType commonType(DataType left, DataType right) {
         if (DataTypes.isNull(right)) {
             return left;
         }
+        if (left.isString() && right.isString()) {
+            if (left == TEXT) {
```

Comment:
Why `TEXT` and not `KEYWORD`?

Reply:
Thought that TEXT is "larger" as double > float, and TEXT is better to be exposed in case the function is used somewhere that we need to check for exact/inexact.